Cargo.toml

```diff
@@ -14,13 +14,13 @@ description = "A Rust version of ttrpc."
 protobuf = { version = "2.0", optional = true }
 bytes = { version = "0.4.11", optional = true }
 libc = { version = "0.2.59", features = [ "extra_traits" ] }
-nix = "0.16.1"
+nix = "^0.20.2"
 log = "0.4"
 byteorder = "1.3.2"
 thiserror = "1.0"
 
 async-trait = { version = "0.1.31", optional = true }
-tokio = { version = "1", features = ["rt", "sync", "io-util", "macros", "time"], optional = true }
+tokio = { version = ">=1.5.1, <1.6.0", features = ["rt", "sync", "io-util", "macros", "time"], optional = true }
```

Comment:
Why we need limit the tokio version so strict? and if we need do this, how about using `version = "~1.5.1"`

Reply:
This was the value produced by `cargo audit fix`, but I agree, your version makes more sense :smile:

Reply:
But the too strict limit will limit the version selection of our user, if they select the tokio@1.13, and we force the tokio version to `>=1.5.1, <1.6.0`, the error: ` failed to select a version for `tokio` will be met.

Reply:
After I have changed the tokio version to 1.13 in examples, I got:
![image](https://user-images.githubusercontent.com/1013713/141753695-74cc1c48-8f53-4bce-baaf-95dcb1076f55.png)

Reply:
Hi @Tim-Zhang - If you change the tokio version in `examples/Cargo.toml`, you also need to do the same to `Cargo.toml` afaik.

ftr, I can run `cargo run --example async-server` successfully using the versions on this PR.

Reply:
Think of the examples as a user's program, if the user update the version of tokio in his/her program, how can he/she do the same to the Cargo.toml in ttrpc-rust?

Reply:
Hi @Tim-Zhang - Ah - I see what you're saying now, but I'm not sure how we resolve this.

| branch | main code | examples | Notes |
|-|-|-|-|
| `master` | `1` | `1.0.1` | To permissive? | 
| this PR (take 1) |  `>=1.5.1, <1.6.0` | `>=1.5.1, <1.6.0` | Safe, but too strict? |
| this PR (take 2) | `~1.5.1` | `~1.5.1` | Safer than `master`, but still too strict? |

Do you think this PR should only update the `nix` version then (so that `tokio = "1"` still for the main code, but maybe `tokio = "~1.5.1"` for the example code)? That is the most "flexible" solution for users of this crate I guess. However, it would allow them to use an old (and potentially insecure) version of tokio.

Reply:
@jodh-intel I think the solution that only update the nix version is ok.

Reply:
@Tim-Zhang - ok. Branch updated.